tuf/tuf.go

```diff
@@ -185,10 +211,19 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey) error
 		return data.ErrInvalidRole{Role: role.Name, Reason: "not a valid delegated role"}
 	}
 	parent := filepath.Dir(role.Name)
-	p, ok := tr.Targets[parent]
-	if !ok {
+
+	// check the parent role
+	if parentRole := tr.keysDB.GetRole(parent); parentRole == nil {
 		return data.ErrInvalidRole{Role: role.Name, Reason: "parent role not found"}
 	}
+
+	// check the parent role's metadata
+	p, ok := tr.Targets[parent]
+	if !ok { // the parent targetfile may not exist yet - if not, then create it
+		p = data.NewTargets()
```

Comment:
Yeah, that would be even better. Maybe a Repo method `CanSign(role string) bool` and as long as the CryptoService has at least one signing key for the role it returns true.